File: BotsSDK/korebotsdklib/src/main/java/kore/botssdk/models/BrandingOverrideHistoryRecentModel.java
package kore.botssdk.models;

import kore.botssdk.utils.StringUtils;

public class BrandingOverrideHistoryRecentModel {
    private int batch_size;
    private String loading_label;
    private Boolean enable;

    public BrandingOverrideHistoryRecentModel updateWith(BrandingOverrideHistoryRecentModel configModel) {
        enable = configModel.enable != null ? configModel.enable : enable;
        loading_label = !StringUtils.isNullOrEmpty(configModel.loading_label) ? configModel.loading_label : loading_label;
        return this;
    }

    public int getBatch_size() {
        return batch_size;
    }

    public String getLoading_label() {
        return loading_label;
    }

    public boolean isEnable() {
        return enable != null ? enable : false;
    }
}

File: BotSdk/botclient/src/main/java/com/kore/network/api/responsemodels/branding/BrandingOverrideHistoryRecentModel.kt
package com.kore.network.api.responsemodels.branding

import com.google.gson.annotations.SerializedName

data class BrandingOverrideHistoryRecentModel(
    @SerializedName("batch_size")
    val batchSize: Int? = 0,
    @SerializedName("loading_label")
    val loadingLabel: String? = null,
    @SerializedName("isEnable")
    val isEnable: Boolean? = false,
)

fun BrandingOverrideHistoryRecentModel.updateWith(configModel: BrandingOverrideHistoryRecentModel?): BrandingOverrideHistoryRecentModel {
    return this.copy(
        isEnable = configModel?.isEnable ?: isEnable,
        batchSize = configModel?.batchSize ?: batchSize,
        loadingLabel = configModel?.loadingLabel?.ifEmpty { loadingLabel } ?: loadingLabel
    )
}
